### PR TITLE
Use mimic for speaker

### PR DESCRIPTION
**Pull request by Flova**
## Proposed changes

- Use mimic3 neural tts instead of espeak
- Refactor speaker node

## Limitations
- Due to limitations of `aplay` the system volume needs to be used.

## Related issues
#129 
https://github.com/bit-bots/bitbots_motion/pull/388

### CODE REVIEW

**Comment by Flova** [comment]
New dependency on `mycroft-mimic3-tts`.

**Review by SammyRamone — APPROVED** [review]
lgtm
nice to have a new voice. maybe we should target porting this to ros-sports after testing it a bit ourselves

File: humanoid_league_speaker/humanoid_league_speaker/speaker.py [hunk 1]
```diff
@@ -19,162 +23,108 @@ def speak(text, publisher, priority=20, speaking_active=True):
...
+        # Callback for parameter changes
+        self.add_on_set_parameters_callback(self.on_set_parameters)
+
+        # Mapping from robot name to voice name
+        self.robot_voice_mapping = {
+            "amy": "en_US/vctk_low",
+            "donna": "en_US/vctk_low",
+            "jack": "en_US/vctk_low",
+            "melody": "en_US/vctk_low",
+            "rory": "en_US/vctk_low"
```
Comment (jaagut) [inline humanoid_league_speaker/humanoid_league_speaker/speaker.py:58]: Do you want to adapt those voices for each robot? I guess, that makes the preloading harder?

Comment (Flova) [reply humanoid_league_speaker/humanoid_league_speaker/speaker.py:58]: For now I didn't change the voices and selected the one I liked the most, but we can definitely look into this in the future.

**Comment by Flova** [comment]
Also tested on robot. Sounds the best at ca. 90% volume.

File: humanoid_league_speaker/humanoid_league_speaker/speaker.py (hunk 1 above)
Comment (Flova) [reply humanoid_league_speaker/humanoid_league_speaker/speaker.py]: I changed them based on gender

**Review by timonegk — APPROVED** [review]
Sounds good on the robot. Please change the volume in set_volume.sh and the motion launch file. 100% sounds really bad, I prefer 70-80%.

**Comment by Flova** [comment]
Maybe we should open an Ansible issue for that?

**Comment by Flova** [comment]
We also need to add the new dependency to the meta requirements before this pr is merged.

**Comment by Flova** [comment]
I opened an issue in the ansible

**Comment by jaagut** [comment]
> Sounds good on the robot. Please change the volume in set_volume.sh and the motion launch file. 100% sounds really bad, I prefer 70-80%.

We have this issue regarding that: bit-bots/bitbots_misc/issues/90

Could we still do it via dynamic reconfigure? Is it necessary?

**Comment by Flova** [comment]
This is not easily possible anymore with the current architecture :( 

We use aplay to play the sound and it only works with the system volume. 
We could adjust the system volume via a ros parameter, but this seems to be not so clean imo.

**Comment by jaagut** [comment]
Yeah, I am also fine with @timonegk suggestion.
If we agree on that, we could close the issue I mentioned above.

**Comment by jaagut** [comment]
> Sounds good on the robot. Please change the volume in set_volume.sh and the motion launch file. 100% sounds really bad, I prefer 70-80%.

I would have only changed the volume to 75% in the motion launch file. I believe the default of the script if no value is given, should still be 100%.

**Comment by jaagut** [comment]
> I would have only changed the volume to 75% in the motion launch file. I believe the default of the script if no value is given, should still be 100%.

Done in bit-bots/bitbots_misc/pull/208

**Pull request merged**